doc/Rules.md

```diff
@@ -1984,3 +1986,20 @@ The configured strong style can be a specific symbol to use ("asterisk",
 "underscore"), or can require that usage be consistent within the document.
 
 Rationale: Consistent formatting makes it easier to understand a document.
+
+## MD051 - No dead hash links within document
+
+Tags: links
+
+Aliases: no-dead-hash-links-within-document
+
+This rule is triggered whenever a dead hash link is found within the document:
```

Comment:
This rule is triggered if a link fragment does not correspond to an heading within the document.

Reply:
It's probably necessary to say how a heading is turned into a link. It's not part of the CommonMark specification and it may be GitHub-specific. Let's link to GitHub documentation if so.

Reply:
Avoid the word "dead" and "slug" if possible, please.